encoding/json/decode.go

```diff
@@ -1266,6 +1268,15 @@ func (d *Decoder) decodeType(valueJSON any, results typeDecodingResults) cadence
 			d.gauge,
 			d.decodeType(obj.Get(typeKey), results),
 		)
+	case "Restriction":
+		// Backwards-compatibility for format <=v0.42.0:
```

Comment:
consider adding a `backwardsCompatible` flag that fails if its set. Create a new `NewBackwardsCompatibleDeccoder` func for that

Reply:
hasn't been determined by the team yet if this is something they can do, but i'll follow up if we do change